### PR TITLE
INSTUI-3028 - TS typing `ui-a11y-content`

### PR DESCRIPTION
**Pull request by ke1k0**

### CODE REVIEW

File: packages/ui-a11y-content/src/ScreenReaderContent/index.tsx [hunk 1]
```diff
@@ -73,8 +79,7 @@ class ScreenReaderContent extends Component<Props> {
     this.props.makeStyles()
   }
 
-  // @ts-expect-error ts-migrate(6133) FIXME: 'prevProps' is declared but its value is never rea... Remove this comment to see the full error message
-  componentDidUpdate(prevProps, prevState, snapshot) {
+  componentDidUpdate() {
     // @ts-expect-error ts-migrate(2722) FIXME: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
```
Comment (ke1k0) [inline packages/ui-a11y-content/src/ScreenReaderContent/index.tsx:83]: I didn't touch these warnings for now, these should be handled when typing the emotion package.

File: packages/ui-react-utils/src/getElementType.ts [hunk 2]
```diff
@@ -37,15 +47,14 @@ import { ComponentType, ReactElement } from 'react'
...
+function getElementType<T extends ComponentWithAsProp>(
+  // TODO: remove |any and as any casts when propTypes are removed
   Component: ComponentType<T> | any,
   props: T,
-  getDefault?: () => string | ReactElement
-) {
-  if (props.as && props.as !== (Component.defaultProps as any).as) {
+  getDefault?: () => AsElementType
+  // TODO: add better typing to the return value
+): any {
```
Comment (ke1k0) [inline packages/ui-react-utils/src/getElementType.ts:56]: It is really hard to type this one, probably needs a bigger refact. I'll open a ticket for it.

File: packages/emotion/package.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
   "license": "MIT",
   "dependencies": {
     "@babel/runtime": "^7.13.10",
-    "@emotion/react": "11.0.0",
+    "@emotion/react": "11.4.0",
```
Comment (ke1k0) [inline packages/emotion/package.json:28]: I copied this from https://github.com/instructure/instructure-ui/pull/655.

File: packages/ui-a11y-content/src/ScreenReaderContent/__tests__/ScreenReaderContent.test.tsx [hunk 4]
```diff
@@ -28,7 +28,6 @@ import { ScreenReaderContent } from '../index'
 
 describe('<ScreenReaderContent />', async () => {
   it('should render the specified tag when `as` prop is set', async () => {
-    // @ts-expect-error ts-migrate(2554) FIXME: Expected 2 arguments, but got 1.
     const subject = await mount(<ScreenReaderContent as="div" />)
     expect(subject.getDOMNode()).to.have.tagName('div')
   })
```
Comment (matyasf) [inline packages/ui-a11y-content/src/ScreenReaderContent/__tests__/ScreenReaderContent.test.tsx:33]: When I remove the `ts-expect-error on line 36 in this file I get this error: `  Property 'hidden' does not exist on type` ...

**Review by matyasf — APPROVED** [review]

File: packages/ui-react-utils/src/getElementType.ts [hunk 5]
```diff
@@ -25,6 +25,16 @@
 import { logWarn as warn } from '@instructure/console'
 import { ComponentType, ReactElement } from 'react'
 
+type AsElementType = keyof JSX.IntrinsicElements | ReactElement | ComponentType
```
Comment (Brailor) [inline packages/ui-react-utils/src/getElementType.ts:28]: I think we will have to move this type to the `shared-types` package once it is merged to master.

**Review by Brailor — APPROVED** [review]

File: packages/ui-a11y-content/src/ScreenReaderContent/__tests__/ScreenReaderContent.test.tsx (hunk 4 above)
Comment (ke1k0) [reply packages/ui-a11y-content/src/ScreenReaderContent/__tests__/ScreenReaderContent.test.tsx:33]: Thanks for pointing it out, forgot to add the OtherHTMLAttributes to the components. Fixed!

File: packages/ui-react-utils/src/getElementType.ts (hunk 5 above)
Comment (ke1k0) [reply packages/ui-react-utils/src/getElementType.ts]: @Brailor I just did, can you please recheck it?

**Review by Brailor — APPROVED** [review]

**Pull request merged**